src/main/java/org/lk/controller/GreetingController.java

```diff
@@ -0,0 +1,46 @@
+package org.lk.controller;
+
+import org.springframework.util.DigestUtils;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController // @Controller
+@RequestMapping("/greeting")
+public class GreetingController {
+
+    @PostMapping("/login")
+    public Greeting authentication(@RequestBody GreetingRequest request) {
+        Greeting obj = new Greeting();
+        Boolean flag = false;
+        Map<String, String> loginAndPassword = new HashMap<>();
```

Comment:
Rewrite this variable as the class field and fill this map in a code block or you can create `public void` method and annotate it with `@PostContruct`